auth_api_key/models/ir_http.py

```diff
@@ -0,0 +1,37 @@
+# -*- coding: utf-8 -*-
+# Copyright 2018 ACSONE SA/NV
+# Copyright 2017 Akretion (http://www.akretion.com).
+# @author Sébastien BEAU <sebastien.beau@akretion.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+import logging
+
+from odoo.http import request
+from odoo import models
+from odoo.exceptions import AccessDenied
+
+
+_logger = logging.getLogger(__name__)
+
+
+class IrHttp(models.AbstractModel):
+    _inherit = "ir.http"
+
+    @classmethod
+    def _auth_method_api_key(cls):
+        headers = request.httprequest.environ
+        api_key = headers.get("HTTP_API_KEY")
+        if api_key:
+            request.uid = 1
+            uid = request.env["auth.api.key"].retrieve_uid_from_api_key(api_key)
```

Comment:
@qgroulard `retrieve_uid_from_api_key`must start with '_'
```suggestion
            uid = request.env["auth.api.key"]._retrieve_uid_from_api_key(api_key)
```